### PR TITLE
Fix(DB/SAI) Into The realm of Shadows issues

### LINKED ISSUE #20107: [DK Starter Chain Quest] Into the Realm of Shadows issues.
### Current Behaviour

The quest can be completed, but:
Acherus Deathchargers are targetable, attackable and killable.
Acherus Deathchargers have half of their hp.

Evidence: [Youtube](https://youtu.be/wPb05KhWvd8)

### Expected Blizzlike Behaviour

Acherus Deathchargers shouldn't be targetable, attackable and killable.
Acherus Deathchargers shouldn't have half of their hp.

How it should work: [Youtube](https://youtu.be/e-j7fkhALic)

### Source

_No response_

### Steps to reproduce the problem

1. Accept the quest (ID: 12687).
2. Do the mission.

### Extra Notes

Before fixing this issue, look at this [report](https://github.com/azerothcore/azerothcore-wotlk/issues/20142).

### AC rev. hash/commit

https://github.com/azerothcore/azerothcore-wotlk/commit/784e80668d5b57640a4c7a4d6e2c7b898e4f804d

### Operating system

Windows 11

### Custom changes or Modules

_No response_

### PR DESCRIPTION
**Pull request by ConorB1745**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [x] Database (SAI, creatures, etc).

Dark Rider of Acherus (id: 28768) was missing some on death scripting that controled the deathcharger (id: 28782). Use smart ai to make the deathcharger rooted in place, emote a message, remove not selectable unitflag to allow player to mount and apply 100% health to ensure deathcharger isn't at 50% health.

Added the emote text to creature_text table for deathcharger.

Added NON_ATTACKABLE and NOT_SELECTABLE unitflags to deathcharger.

Added SmartAI script to deathcharger, so that when a passenger enters the horse unroots.

## Issues Addressed:

- Closes #20107 

## SOURCE:

The changes have been validated through:
- [x] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [x] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

Wotlk classic video of the quest:
https://youtu.be/lWF-tDzBPlw 

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [x] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Rorschach91** [comment]
I have tested this PR and It seems that It works.
It can be reviewed.

**Comment by Rorschach91** [comment]
This PR should also close this report, can you add it? 
https://github.com/azerothcore/azerothcore-wotlk/issues/2611

**Comment by ConorB1745** [comment]
> This PR should also close this report, can you add it? #2611

I tested the description of issue #2611 against the changes in this one. I was not able to replicate the first point of a riderless horse. When fighting the "Black Rider of Acherus" (id=28768) the horse does attack aswell which is blizzlike behaviour however it does alot more damage than I was able to replicate in cata classic.

Cata classic damage log (around 10-13 damage):
testing_archerus_deathcharger_on_cata_classic

Azeroth Core damage log (149-160):
testing_archerus_deathcharger_on_azerothcore

File: data/sql/updates/pending_db_world/rev_1728239921994699100.sql [hunk 1]
```diff
@@ -0,0 +1,20 @@
+UPDATE `creature_template` SET `unit_flags` = 33587202 WHERE (`entry` = 28782);
+
+INSERT INTO creature_text
+(CreatureID, GroupID, ID, `Text`, `Type`, `Language`, Probability, Emote, Duration, Sound, BroadcastTextId, TextRange, comment)
+VALUES(28782, 0, 0, '%s rears up, beckoning you to ride it.', 16, 0, 100.0, 0, 0, 0, 29069, 0, 'Archerus Deathcharger');
```
Comment (Helias) [inline data/sql/updates/pending_db_world/rev_1728239921994699100.sql:5]: Before an insert you have to put a DELETE, Moreover, please add the backtick in all columns and table name

Comment (ConorB1745) [reply data/sql/updates/pending_db_world/rev_1728239921994699100.sql:6]: I've implemented the changes mentioned here.

**Review by Helias — APPROVED** [review]

**Pull request merged**